ahnlich/ai/src/engine/store.rs

```diff
@@ -0,0 +1,93 @@
+use ahnlich_types::ai::AIModel;
+use ahnlich_types::ai::AIStoreInfo;
+use ahnlich_types::ai::AIStoreType;
+use ahnlich_types::keyval::StoreInput;
+use ahnlich_types::keyval::StoreName;
+use ahnlich_types::metadata::MetadataKey;
+use flurry::HashMap as ConcurrentHashMap;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::HashMap as StdHashMap;
+use std::collections::HashSet as StdHashSet;
+use std::mem::size_of_val;
+use std::sync::atomic::AtomicBool;
+use std::sync::Arc;
+
+use crate::error::AIProxyError;
+/// Contains all the stores that have been created in memory
+#[derive(Debug)]
+pub struct AIStoreHandler {
+    /// Making use of a concurrent hashmap, we should be able to create an engine that manages stores
+    stores: AIStores,
+    pub write_flag: Arc<AtomicBool>,
+}
+
+pub type AIStores = Arc<ConcurrentHashMap<StoreName, Arc<AIStore>>>;
+
+impl AIStoreHandler {
```

Comment:
The StoreHandler has to have ownership of some DBClient that would be used to make the calls